libs/cli/src/test/java/org/elasticsearch/cli/internal/JsonPrintWriterTests.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cli.internal;
+
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+import static com.carrotsearch.randomizedtesting.generators.RandomStrings.randomAsciiAlphanumOfLengthBetween;
+
+public class JsonPrintWriterTests extends ESTestCase {
```

Comment:
Can you add a test with printing a stack trace from an exception? I suspect we will get multiple json objects, one per line of the stack trace. If so, we should re-plumb where exceptions are handled for the cli to work a little different. Similar question with the help message output, or anywhere we grab the PrintWriter directly.

Reply:
Yes, usage of `Exception#printStackTrace` would result in stracktraces being scattered across multiple json objects. Looking at the code, I also noticed everything but UserException and OptionException bubble through uncaught and wouldn't be logged as JSON at all. I'll follow up on that.

Reply:
https://github.com/elastic/elasticsearch/pull/104055/commits/c191edce596b662e003c2652e89c2c6c3392c78e adds support to write a stacktrace into a single JSON object. Though, that also requires to catch all exceptions before exciting `main`.

Reply:
https://github.com/elastic/elasticsearch/pull/104055/commits/d1e75e0f88168de5e7bc41e0ed5cd31be1cc7545 verifies the behavior with respect to the option parser / printing help.